azurerm/internal/services/loganalytics/resource_arm_log_analytics_datasource_linux_syslog.go

```diff
@@ -0,0 +1,236 @@
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"emerg",
+						"alert",
+						"crit",
+						"err",
+						"warning",
+						"notice",
+						"info",
+						"debug",
+					}, false),
+				},
+			},
+
+			"syslog_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"auth",
+					"authpriv",
+					"cron",
+					"daemon",
+					"ftp",
+					"kern",
+					"local0",
+					"local1",
+					"local2",
+					"local3",
+					"local4",
+					"local5",
+					"local6",
+					"local7",
+					"lpr",
+					"mail",
+					"news",
+					"syslog",
+					"user",
+					"uucp",
+				}, false),
+			},
+		},
+	}
+}
+
+type dataSourceLinuxSyslog struct {
+	Name       string                          `json:"syslogName"`
+	Severities []dataSourceLinuxSyslogSeverity `json:"syslogSeverities"`
+}
+
+type dataSourceLinuxSyslogSeverity struct {
+	Severity string `json:"severity"`
+}
+
+func resourceArmLogAnalyticsDataSourceLinuxSyslogCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.DataSourcesClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	workspaceName := d.Get("workspace_name").(string)
+
+	if d.IsNewResource() {
+		resp, err := client.Get(ctx, resourceGroup, workspaceName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("failed to check for existing Log Analytics Data Source Linux Syslog %q (Resource Group %q / Workspace: %q): %+v", name, resourceGroup, workspaceName, err)
+			}
+		}
+
+		if resp.ID != nil && *resp.ID != "" {
+			return tf.ImportAsExistsError("azurerm_log_analytics_datasource_linux_syslog", *resp.ID)
+		}
+	}
+
+	param := operationalinsights.DataSource{
+		Kind: operationalinsights.LinuxSyslog,
+		Properties: &dataSourceLinuxSyslog{
+			Name:       d.Get("syslog_name").(string),
+			Severities: expandLogAnalyticsDataSourceLinuxSyslogSeverities(d.Get("severities").(*schema.Set).List()),
+		},
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, workspaceName, name, param); err != nil {
+		return fmt.Errorf("failed to create Log Analytics DataSource Linux Syslog %q (Resource Group %q / Workspace: %q): %+v", name, resourceGroup, workspaceName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, workspaceName, name)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve Log Analytics Data Source Linux Syslog %q (Resource Group %q / Workspace: %q): %+v", name, resourceGroup, workspaceName, err)
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("cannot read Log Analytics Data Source Linux Syslog %q (Resource Group %q / Workspace: %q) ID", name, resourceGroup, workspaceName)
```

Comment:
and
```suggestion
		return fmt.Errorf("reading Log Analytics Data Source Linux Syslog %q (Resource Group %q / Workspace: %q) ID", name, resourceGroup, workspaceName)
```

as the error or failed is implied? YDYT?